packages/wrangler/src/dev.ts

```diff
@@ -561,216 +700,158 @@ export async function startDev(args: StartDevOptions) {
 			args.config ||
 			(args.script && findWranglerToml(path.dirname(args.script)));
 
-		// The ProxyWorker will have a stable host and port, so only listen for the first update
-		void devEnv.proxy.ready.promise.then(({ url }) => {
-			if (args.onReady) {
-				args.onReady(url.hostname, parseInt(url.port));
+		const authHook: AsyncHook<CfAccount, [Pick<Config, "account_id">]> = async (
+			config
+		) => {
+			const hotkeysDisplayed = !!unregisterHotKeys;
+			let accountId = args.accountId;
+			if (!accountId) {
+				unregisterHotKeys?.();
+				accountId = await requireAuth(config);
+				if (hotkeysDisplayed) {
+					assert(devEnv !== undefined);
+					unregisterHotKeys = registerDevHotKeys(
+						Array.isArray(devEnv) ? devEnv[0] : devEnv,
+						args
+					);
+				}
 			}
+			return {
+				accountId,
+				apiToken: requireApiToken(),
+			};
+		};
 
-			if (
-				(args.enableIpc || !args.onReady) &&
-				process.send &&
-				typeof vitest === "undefined"
-			) {
-				process.send(
-					JSON.stringify({
-						event: "DEV_SERVER_READY",
-						ip: url.hostname,
-						port: parseInt(url.port),
-					})
-				);
+		if (Array.isArray(configPath)) {
```

Comment:
I feel like we could clean things up if we just treated all scenarios as multi-worker. So instead of having a separate `MultiworkerRuntimeController`, just modify the `LocalRuntimeController` to handle multi-workers, and then cast the configPath to an array in the yargs definition. 

```ts
const configs = Array.isArray(configPath) ? configPath : [configPath];

// Now no need to handle the single worker case. LocalRuntimeController assumes multi.
const runtime = new LocalRuntimeController(configs.length);
```

Or is there a reason to keep the single worker code path?

Reply:
I went down this path in [Merge Local & Multi controllers](https://github.com/cloudflare/workers-sdk/pull/7251/commits/d632e70131e553a3814aceccf59d8ea10e55f84f), but I'm starting to get a bit concerned about the potential for disrupting the normal `dev` flow, and I think it might end up being more complexity than it's worth. The single worker dev flow has two runtimes, remote & local, and the remote runtime doesn't (and will never) work for the multiworker use case. Because of that there's some differences in how these flows are set up (as well as the disabling of the dev registry in the multiworker flow, which involves disabling the registry as well as a change to how DOs & entrypoints are registered). It's not that much duplicated code, and I think it's actually clearer to keep them separate, and easier to reason about. What do you think?

Reply:
Yeah, that sounds fair enough. Just for clarity, I assume the remote runtime will never be able to do this because it is always deploying a single worker.